components/provisioner/internal/gardener/shoot_reconciler.go

```diff
@@ -91,6 +103,14 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, err
 	}
 
+	seed := getSeed(shoot)
+	if seed != "" {
+		err := r.enableAuditLogs(&shoot, r.auditLogsCMName, seed)
+		if err != nil {
+			log.Errorf("Failed to enable audit logs for %s shoot: %s", shoot.Name, err.Error())
+		}
+	}
```

Comment:
I think this code should be called after checking if `shouldReconcileShoot`, so that each Provisioner handles only Shoots from its database.